### ISSUE TITLE
User lost if he/she has no password - pilot user feedback

### ISSUE THREAD

**Issue by franck-boullier** [unee-t/frontend#429]
# The problem:

Several pilot users are reporting that they are not able to access Unee-T because they have no password.

# The solution:

Find a way to guide the user who has no password so he/she uses the 'reset password'  mechanism  if the user does NOT know his/her password to access Unee-T

@kiatlim any suggestion/recommendation?

**Comment by kiatlim** [comment]
One way is to reword the "Forgot password?" to:
"Click here to reset your password if you have forgotten or do not know your password." and shift the Login button down to the next line.

I'll provide a screenshot for reference. 

Also, for users who have login to Unee-T via the invitation magic link that logs them in magically, we can reuse the prompt/design of "Verify your email address" to ask them to "Set your password" which leads them into the Account screen.

**Comment by kaihendry** [comment]
We should have a FAQ that answers obvious search terms like:
* [$APP_NAME lost password](https://www.google.com.sg/search?q=uneet+lost+password)
* [$APP_NAME reset password](https://www.google.com.sg/search?q=uneet+reset+password)

e.g. for my own app https://www.google.com.sg/search?q=webconverger+lost+password

**Comment by yvonnechia** [comment]
> > I'll provide a screenshot for reference.
> 
> > we can reuse the prompt/design of "Verify your email address" to ask them to "Set your password" which leads them into the Account screen.
> 
> @kiatlim and @michkwt I think this is still pending



"Forgot password?" has been reworded to "Reset password" at login screen. 

When user has first been invited into a case, he/she will be prompted by the alert panel at the top to create a new password. 

![screen shot 2018-10-11 at 4 24 32 pm](https://user-images.githubusercontent.com/43780040/46790761-3ab97a80-cd72-11e8-8472-d729449edb06.png)

**Comment by franck-boullier** [comment]
I like that option @yvonnechia 👍

**Comment by kiatlim** [comment]
@nbiton @sophieyang0813 FYI. 
@franck-boullier I guess this can go into implementation?

**Comment by franck-boullier** [comment]
@kiatlim 
We were discussing this with @nbiton and @kaihendry today and we want to implement something more obvious for the users who alredy have an account.

# The Scenario:

- A user has a Unee-T account
- The user is invited to or assigned to a case.
- The User receives a Unee-T email notification
- The user clicks on the link included in the email.
- The user is NOT already logged in to Unee-T on the device he's currently using.

# What happens today:

The user is sent to the login page:
![image](https://user-images.githubusercontent.com/31331637/55208127-fafb2800-5217-11e9-9e90-342cb897b757.png)

# The issue with the current behavior:

- This page is confusing for many user.

Examples: 
- This is the first time this user receives a notification.
- The user has not logged in to Unee-T in a while.
- The user is using a different device.

## Why is this confusing:

- The page title is "Sign In".
- The user has no information about the login name he/she should use.
- There is message "Don't have an account? Sign Up for free" ---> this message is misleading in this scenario: 
  - We know the user already has an account.
  - We know that the user is seeing this page only because he needs to log in to his/her Unee-T account.

# Proposed solution:

IF we send the user a notification, then we know the user has an account.
IF the user is not already logged in to Unee- T on the device he/she uses, THEN we send the user to a different page (NOT the default first time account creation/login for a user shown above).

This different page should:
- Highlight the fact that the user ALREADY has a Unee-T account
- Pre-populate the login name for the user (we know this since we are sending this the the user's email address)
- Ask for the password associated to that email address.
- Make the "Reset your Password" option much more obvious/Visible than what it currently is.

# What needs to be done:

@kiatlim and Dominic, can you suggest a design for this new page?
Feel free to ping me, should you have any question on this!

# Why is this critical?

Today we have a lot of users that are dropping off and not login to Unee-T when they receive a notification message.
We need to improve that to make sure we have the minimum amount of friction and to make it very easy/obvious for users to access and use Unee-T.

**Comment by kiatlim** [comment]
@franck-boullier @ongdominic @nbiton @kaihendry I think there's a few possible solutions here.

1. Extend the current login screen such that when the user clicks on the link and land on the login screen for existing users, his email is pre-populated, instead of saying "Sign in", the wording says something contextual like "Sign in to view your case" or "Sign in to view your unit" etc. - _user still requires password._

2. If they are already an existing users, the link in the email is actually a magic link that signs the user in automagically and directs him to the page itself. No email and password to figure out but could have security issue.

3. The link in the email leads user to the page which he is suppose to view e.g. case details but he sees a [Sign in to comment] or [Sign in] to prompt that he has to login before he could do anything. (Something like Github) - but this could lead to having a public view of the details. User still needs to sign in with email and password but at least he sees the value of having to do prior. 

4. The link in the email leads the user to a simple page that ask for his email address. He fills in the email address and receives a magic link. He clicks on the magic link in the email and boom, he's logged in to view the case automagically. No fiddling with passwords. 

If we checked that the password of the user is not being set yet, then this reminder may come in handy? 

![image](https://user-images.githubusercontent.com/641679/55271078-54c82480-52e2-11e9-8628-a174ca1d8aaf.png)

5. The link in the email leads the user to a simple page that just ask for his email address. Fills it in and he's in the case. No further login needed.

What do you think?

**Comment by franck-boullier** [comment]
Hi @kiatlim thanks for the suggestions.

# Facilitate User Login to Unee-T when email is received:

TBH I'm not convinced by the idea that the only security mechanism we have is to ask for an email address: this is way to public and would make this extremely insecure.

I think that if the user is not already logged in, then we should ask for a password always. This means that the "Forgot password option" should also be very obvious as it is very likely some user will not know/remember their passwords since they won't be using Unee-T on a daily basis.

The main issue I have with the magic link option is that if the email is forwarded to someone else, then we have no guarantee to be sure that the user that logged in is the actual initial person that was expected to log in ---> we're back to square 1: we need to have some sort of authentication mechanism like password or OTP.

Based on these your Idea 3 is my favorite:
> The link in the email leads user to the page which he is suppose to view e.g. case details but he sees a [Sign in to comment] or [Sign in] to prompt that he has to login before he could do anything. (Something like Github) - but this could lead to having a public view of the details. User still needs to sign in with email and password but at least he sees the value of having to do prior.

This looks to be an acceptable compromise.
We can probably make it so that the user can't scroll up or go to any other page in this scenario ---> this will limit the amount of data we expose and still give enough context and incentive to get the user to sign in.

Can you think of a design for this?

**Comment by kiatlim** [comment]
@franck-boullier Yeah sure we can work out a design for this on the case conversation page and perhaps unit detail page.

What do you think about idea 4? it's actually using email as an OTP. Since we cannot guarantee now that user will always have a mobile phone to their account, we use their email as the OTP. 

When they receive the link and click on it, it leads them to a page that ask them to enter their email. After entering their email, they receive another email thats with a magic link to view the case. 

We will also look at further at improving the Reset Password / Forgot Password on the Login screen to make it more obvious.

**Comment by franck-boullier** [comment]
> What do you think about idea 4? 

My main gripe with that is that an email is a very public piece of information (I know yours) as opposed to OTP and Passwords which are, by definition, not public.

Also the "magic link" is an issue if the email is forwarded to someone else ---> that other person could log in as the user who was supposed to receive the initial email ---> this is a problem.

> We will also look at further at improving the Reset Password / Forgot Password on the Login screen to make it more obvious.

Thanks!

**Comment by ongdominic** [comment]
Designs
https://www.figma.com/file/SgLcXdmBih1JxVq1lupMiPtr/Unee-T-Designs?node-id=2223%3A23

Prototype
https://www.figma.com/proto/SgLcXdmBih1JxVq1lupMiPtr/Unee-T-Designs?node-id=870%3A83&scaling=scale-down

**Comment by franck-boullier** [comment]
@ongdominic see my comments on Figma

**Comment by ongdominic** [comment]
@nbiton [Here](https://www.figma.com/file/SgLcXdmBih1JxVq1lupMiPtr/Unee-T-Designs?node-id=2223%3A23) is the link to the designs that solves this issue.

I have also created a prototype that goes thru the 2 different flows. 
[Prototype] (https://www.figma.com/proto/SgLcXdmBih1JxVq1lupMiPtr/Unee-T-Designs?node-id=870%3A83&scaling=scale-down)

**Comment by franck-boullier** [comment]
Because Users invited to a case are now receiving a magic link and are automatically logged in in most of the cases this is not a critical issue anymore.

**Comment by ongdominic** [comment]
This is the solution that we are implementing for users who have not set their password to remind the users to set their password. 

if the user has not set his password he will see this red section of the screen. 

<img width="375" alt="Invitee" src="https://user-images.githubusercontent.com/46714451/59335012-4a88c480-8d2e-11e9-9410-f71bb210be26.png">

Details of the design are in [figma](https://www.figma.com/file/SgLcXdmBih1JxVq1lupMiPtr/Unee-T-Designs?node-id=870%3A555)

**Comment by nbiton** [comment]
IMO, we'd need to verify the user's identity again before letting them assign a password to their account. This would provide better security for users and help prevent cases of account "hijacking".
Therefore, I suggest that clicking the "Create a new password" shown in the design mock would result in a flow similar to the password reset flow we already have:
1. The backend will send a password reset link via email to this user's registered email address.
2. A message will appear on the user's screen that instructs to look in the email inbox for the reset link.
3. Clicking on the email link would lead the user to the reset password view we already use.

This will ensure account security in a slightly better way, and should allow us to use exiting components which can save time to implement it.

**Comment by franck-boullier** [comment]
I agree with @nbiton 's latest comment, let' re-verify the user identity again.
@ongdominic can you update the design so we can all be on the same page on what the flow should be.

**Comment by ongdominic** [comment]
Here is the new flow of creating a new password for users who have not created a password before with an additional step to re-verify a user

<img width="806" alt="Screenshot 2019-06-18 at 8 09 09 PM" src="https://user-images.githubusercontent.com/46714451/59680860-07829180-9205-11e9-933b-0501eb65b5a3.png">

Details are in [figma](https://www.figma.com/file/SgLcXdmBih1JxVq1lupMiPtr/Unee-T-Designs?node-id=3191%3A5488)
